cmd/deploy/deploy.go

```diff
@@ -348,3 +373,136 @@ func getProxyHandler(name, token string, clusterConfig *rest.Config) (http.Handl
 	return handler, nil
 
 }
+
+func (dc *deployCommand) waitUntilDeployed(ctx context.Context, opts *Options) error {
+	labelSelector := fmt.Sprintf("%s=%s", model.DeployedByLabel, opts.Name)
+	c, _, err := okteto.GetK8sClient()
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(1 * time.Second)
+	to := time.NewTicker(opts.Timeout)
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("pipeline '%s' didn't finish after %s", opts.Name, opts.Timeout.String())
+		case <-ticker.C:
+			isReady := true
+			dList, err := deployments.List(ctx, okteto.Context().Namespace, labelSelector, c)
+			if err != nil {
+				return err
+			}
+			for _, dep := range dList {
+				if dep.Status.ReadyReplicas != dep.Status.Replicas {
+					isReady = false
+				}
+			}
+			sfsList, err := statefulsets.List(ctx, okteto.Context().Namespace, labelSelector, c)
+			if err != nil {
+				return err
+			}
+			for _, sfs := range sfsList {
+				if sfs.Status.ReadyReplicas != sfs.Status.Replicas {
+					isReady = false
+				}
+			}
+
+			if !isReady {
+				continue
+			}
+			return nil
+		}
+	}
+}
+
+func (dc *deployCommand) showEndpoints(ctx context.Context, opts *Options, spinner *utils.Spinner) error {
+	spinner.Update("retrieving endpoints")
+	pods, err := getDeployPodNames(ctx, opts)
```

Comment:
@pchico83 Do we need to retrieve all the pods to get the endpoints? Would there be any case in which using the `okteto deploy` command would not include `dev.okteto.com/deployed-by` labels in ingresses and services?

Reply:
Minor thing: If you convert this list of pods in a map[string]bool, you can save a loop (it is a 4th level loop) in `endpoints.isOnPodList` function as you can use the `refName` as key to look for the pod

Reply:
@ifbyol @jLopezbarb yes, I would just query the ingress by label. We have already waited for deployments/statefulsets to be available before